compiler/AST/ForallStmt.cpp

```diff
@@ -516,6 +547,45 @@ ForallStmt* ForallStmt::fromForLoop(ForLoop* forLoop) {
   return result;
 }
 
+// "Undo" the tuple for a zippered 'fs'.
+static void destructZipperedIterables(ForallStmt* fs, SymExpr* dataExpr) {
+  CallExpr* zipcall = toCallExpr(getDefOfTemp(dataExpr));
+  INT_ASSERT(zipcall->isPrimitive(PRIM_ZIP));
+  for_actuals(actual, zipcall)
+    fs->iteratedExpressions().insertAtTail(actual->remove());
+
+  // Ensure removing zipcall is OK.
+  CallExpr* move = toCallExpr(zipcall->parentExpr);
+  INT_ASSERT(move->isPrimitive(PRIM_MOVE));
+  move->remove();
+}
+
+ForallStmt* ForallStmt::fromReduceExpr(VarSymbol* idx, SymExpr* dataExpr,
+                                       ShadowVarSymbol* svar,
+                                       bool zippered, bool requireSerial)
+{
+  Symbol* ata = new_IntSymbol(-7, INT_SIZE_64); // atavism, ignored
```

Comment:
wha-huh? Why -7? Maybe 0 would be more reasonable, even if it's not used. And, if it's not used, how do you know an `int` will do? Could you use `gDummyReturnValue` e.g.?

Reply:
I added, as a follow-up item for this PR, the removal of this arg altogether.

Meanwhile I am using -7 in order to draw attention to this ridiculous-ness.
(Yes I am serious.)